### PR TITLE
Real-time json index

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
## Description
Add json index support to mutable (real-time) segments
Add a new quickstart for real-time json index on meetup RSVP events

Example query: Find events that are in the fitness group
`select json_extract_scalar(event, '$.event_name', 'STRING') from meetupRsvp where json_match("group", 'group_topics.topic_name=''Fitness''') limit 10`

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -233,8 +233,8 @@ public void deleteSegmentFile() {
   private final SegmentVersion _segmentVersion;
   private final SegmentBuildTimeLeaseExtender _leaseExtender;
   private SegmentBuildDescriptor _segmentBuildDescriptor;
-  private StreamConsumerFactory _streamConsumerFactory;
-  private StreamPartitionMsgOffsetFactory _streamPartitionMsgOffsetFactory;
+  private final StreamConsumerFactory _streamConsumerFactory;
```
Comment (fx19880617) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:236]: do we need final for these ?

File: pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpJsonStream.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    try {
+      _client = ClientManager.createClient();
+      _keepPublishing = true;
+
+      _client.connectToServer(new Endpoint() {
+        @Override
+        public void onOpen(Session session, EndpointConfig config) {
+          session.addMessageHandler(new MessageHandler.Whole<String>() {
+            @Override
+            public void onMessage(String message) {
```
Comment (fx19880617) [inline pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpJsonStream.java:67]: can we extend MeetupRsvpStream and override a method to generate the msg to produce?

**Review by fx19880617 — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:236]: Hmm, seems these are changed by the IDE auto-formatter

File: pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpJsonStream.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpJsonStream.java:67]: Good point, let me add that

**Pull request merged**